nixos/modules/programs/command-not-found/command-not-found.pl

```diff
@@ -38,6 +39,9 @@
         print STDERR <<EOF;
 The program ‘$program’ is currently not installed. You can install it by typing:
   nix-env -iA nixos.$package
+
+Or run it once with:
+  nix-shell -p $package --run $quoted_command
```

Comment:
that should be `nix run` then, not `nix-shell`

Reply:
`nix run` is "experimental" and may even disappear behind a feature flag in the next release.

Reply:
yes, I've seen that change. That's really backwards. `nix-shell` is causing lots of trouble for just running applications because of the hooks, now we have a solution, and then it would "disappear" again :confused:

Reply:
Yes, I agree. But we shouldn't recommend it until that problem is solved.

Reply:
Where is putting `nix run` behind a feature flag discussed? I find it very useful.

Reply:
https://github.com/NixOS/nix/commit/aabf5c86c9df1b4e1616a4cf06ee14a6edf2e5e1